jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -102,8 +95,56 @@ static Path getExplodedWarDirectory(Project project) {
     return project.getBuildDir().toPath().resolve(ProjectProperties.EXPLODED_WAR_DIRECTORY_NAME);
   }
 
+  private static EventHandlers makeEventHandlers(
+      Project project, Logger logger, SingleThreadedExecutor singleThreadedExecutor) {
+    LogEventHandlerBuilder logEventHandlerBuilder =
+        (isProgressFooterEnabled(project)
+                ? LogEventHandlerBuilder.rich(singleThreadedExecutor)
+                : LogEventHandlerBuilder.plain(singleThreadedExecutor).progress(logger::lifecycle))
+            .lifecycle(logger::lifecycle);
+    if (logger.isDebugEnabled()) {
```

Comment:
Sorry I'm stuck on this. And I don't even know if we should handle this. So stop me if I should be stopped.
1. The logger implementations are already filtering the log out based on the log level. So should we do additional filtering here? Presumably what we're doing here can lead to one less call for every log message that needs to be ignored.
2. If someone dynamically changes the log level to a different level, we're kind of stuck in this implementations.

Would it make sense to just pass the supplier in unconditionally and let the log systems's mechanisms handle the filtering according to level?
```
.debug(logger::debug) // also this double debug thing is confusing
.debug(logger::info)
.warn...
.build()
```

Reply:
Yes, the Gradle `Logger` implementation will filter calls to the individual `Logger` methods based on the verbosity chosen by the user, but this API is on the Jib Core side. Each `LogEventHandlerBuilder` method is attaching a `Logger` method to the corresponding `LogEvent.Level` that should forward to the `Logger` method. `Logger` then can do its logging based on the user choice.

The double debug was a typo - will fix.

Reply:
Ah yea right, that makes sense.

Reply:
oh yeah okay, I didn't realize it had that effect.